### PR TITLE
fix(renderer): load UI state before Keychain authentication

### PR DESCRIPTION
**Pull request by fbricon**
On macOS initial startup from DMG, UI was not properly initialized until after Keychain authentication completed. Window showed with black background and missing navigation labels while waiting for user to authenticate.

Fix consists in keep showing the splash screen until colors are loaded from the
main process, preventing a flash of unstyled content. Guard
AppNavigation rendering until iconWithTitle config is resolved.

This is a backport of https://github.com/openkaiden/kaiden/pull/2104
I did not manage to reproduce Kaiden's issue in PD, but Kaiden needs this change done in PD first.

### CODE REVIEW

File: packages/main/src/plugin/index.ts [hunk 1]
```diff
@@ -527,11 +527,9 @@ export class PluginSystem {
       container.bind<Directories>(Directories).to(LegacyDirectories).inSingletonScope();
     }
     container.bind<StatusBarRegistry>(StatusBarRegistry).toSelf().inSingletonScope();
-    container.bind<SafeStorageRegistry>(SafeStorageRegistry).toSelf().inSingletonScope();
-
-    const safeStorageRegistry = container.get<SafeStorageRegistry>(SafeStorageRegistry);
-    notifications.push(...(await safeStorageRegistry.init()));
 
+    // Initialize ConfigurationRegistry and ColorRegistry BEFORE SafeStorageRegistry
+    // to avoid blocking on Keychain auth during initial UI rendering
```
Comment (benoitf) [inline packages/main/src/plugin/index.ts]: here it does not do any init at all, it's the binding so the comments look invalid

it should be where init() methods are called, before there is no usage

File: packages/main/src/plugin/index.ts [hunk 2]
```diff
@@ -542,10 +540,47 @@ export class PluginSystem {
...
+    this.ipcHandle('colorRegistry:listColors', async (_listener, themeId: string): Promise<ColorInfo[]> => {
+      return colorRegistry.listColors(themeId);
+    });
+
+    this.ipcHandle('colorRegistry:isDarkTheme', async (_listener, themeId: string): Promise<boolean> => {
+      return colorRegistry.isDarkTheme(themeId);
+    });
+
+    // Initialize SafeStorageRegistry after ColorRegistry to avoid blocking UI color loading
+    container.bind<SafeStorageRegistry>(SafeStorageRegistry).toSelf().inSingletonScope();
```
Comment (benoitf) [inline packages/main/src/plugin/index.ts]: there is no need to move the declaration of the bind

**Review by benoitf — CHANGES_REQUESTED** [review]
there should not be fetch call on stores

**Review by vancura — APPROVED** [review]

File: packages/renderer/src/Loader.svelte [hunk 3]
```diff
@@ -6,15 +6,23 @@ import App from './App.svelte';
...
 
 let loadingSequence: NodeJS.Timeout;
 
 let extensionsStarterChecker: NodeJS.Timeout;
 
+const unsubscribeColors = colorsInfos.subscribe(colors => {
+  if (colors && colors.length > 0) {
+    colorsLoaded = true;
+  }
+});
```
[the remark this answers is not in the document]
Comment (fbricon) [reply packages/renderer/src/Loader.svelte:24]: tried the tryAction approach and it broke tests because the test's mock window doesn't connect dispatchEvent to addEventListener. This is a low-risk edge case (the redirect only fires for extension install deeplinks, and colors load within milliseconds).

**Comment by fbricon** [comment]
@benoitf used a new approach where the UI waits for proper events before showing components.

**Comment by bmahabirbu** [comment]
Probably needs a rebase to fix test failure. Unrelated to the changes

**Comment by bmahabirbu** [comment]
@fbricon any luck on this one?
